packages/expo-cli/src/credentials/provider/ios.ts

```diff
@@ -0,0 +1,135 @@
+import { SetupIosProvisioningProfile } from '../views/SetupIosProvisioningProfile';
+import { SetupIosDist } from '../views/SetupIosDist';
+import { runCredentialsManager } from '../route';
+import { Context } from '../context';
+import { credentialsJson } from '../local';
+import { CredentialsProvider } from './provider';
+
+export interface iOSCredentials {
+  provisioningProfile: string;
+  distributionCertificate: {
+    certP12: string;
+    certPassword: string;
+  };
+}
+
+interface Options {
+  projectName: string;
+  accountName: string;
+  bundleIdentifier: string;
+}
+
+export class iOSCredentialsProvider implements CredentialsProvider {
+  public readonly platform = 'ios';
+  private readonly ctx = new Context();
+  private credentials?: iOSCredentials;
+
+  constructor(private projectDir: string, private options: Options) {}
+
+  get projectFullName(): string {
+    const { projectName, accountName } = this.options;
+    return `@${accountName}/${projectName}`;
+  }
+
+  public async init() {
+    await this.ctx.init(this.projectDir);
+  }
+
+  public async hasRemote(): Promise<boolean> {
+    const distCert = await this.ctx.ios.getDistCert(
+      this.projectFullName,
+      this.options.bundleIdentifier
+    );
+    const provisioningProfile = await this.ctx.ios.getProvisioningProfile(
+      this.projectFullName,
+      this.options.bundleIdentifier
+    );
+    return !!(distCert && provisioningProfile);
+  }
+
+  public async hasLocal(): Promise<boolean> {
+    if (!(await credentialsJson.exists(this.projectDir))) {
+      return false;
+    }
+    try {
+      await credentialsJson.readAndroid(this.projectDir);
```

Comment:
Android?

Reply:
copy-paste
it worked when I tested it becuase I already had both ios and android credentials